packages/lesswrong/components/forumEvents/givingSeason/EditElectionCandidate.tsx

```diff
@@ -0,0 +1,239 @@
+  { value: "givingSeason25", label: "EA Giving Season 2025" },
+];
+
+const EditElectionCandidate = ({classes}: {
+  classes: ClassesType<typeof styles>,
+}) => {
+  const {flash} = useMessages();
+  const navigate = useNavigate();
+  const {openDialog} = useDialog();
+  const {params} = useLocation();
+  const candidateId = params.id;
+  const isNewForm = candidateId === "new";
+
+  const currentUser = useCurrentUser();
+
+  const { document: existingCandidate, loading: loadingCandidate } = useSingle({
+    collectionName: "ElectionCandidates",
+    fragmentName: "ElectionCandidateBasicInfo",
+    documentId: candidateId,
+    skip: isNewForm,
+  });
+
+  const { create } = useCreate({
+    collectionName: "ElectionCandidates",
+    fragmentName: "ElectionCandidateBasicInfo",
+  });
+
+  const { mutate } = useUpdate({
+    collectionName: "ElectionCandidates",
+    fragmentName: "ElectionCandidateBasicInfo",
+  });
+
+  const { setCaughtError, displayedErrorComponent } = useFormErrors();
+
+  const form = useForm({
+    defaultValues: {
+      electionName: existingCandidate?.electionName ?? "givingSeason25",
+      name: existingCandidate?.name ?? "",
+      logoSrc: existingCandidate?.logoSrc ?? "",
+      href: existingCandidate?.href ?? "",
+      description: existingCandidate?.description ?? "",
+      tagId: existingCandidate?.tagId ?? null,
+    },
+    onSubmit: async ({ formApi }) => {
+      try {
+        let result: ElectionCandidateBasicInfo;
+
+        if (isNewForm) {
+          const { data } = await create({ data: formApi.state.values });
+          result = data?.createElectionCandidate.data;
+        } else {
+          const { data } = await mutate({
+            selector: { _id: candidateId },
+            data: formApi.state.values,
+          });
+          result = data?.updateElectionCandidate.data;
+        }
+
+        flash("Success");
+        navigate({ pathname: "/admin/election-candidates" });
+        setCaughtError(undefined);
+      } catch (error) {
+        setCaughtError(error);
+      }
+    },
+  });
+
+  const deleteCallback = useCallback(() => {
+    if (!isNewForm) {
+      openDialog({
+        name: "DeleteElectionCandidateDialog",
+        contents: ({onClose}) => <DeleteElectionCandidateDialog onClose={onClose} candidateId={candidateId} />
+      });
+    }
+  }, [isNewForm, openDialog, candidateId]);
+
+  if (!userIsAdmin(currentUser)) {
+    return (
+      <Error404 />
+    );
+  }
+
+  if (!isNewForm && loadingCandidate) {
+    return <Loading />;
+  }
+
+  return (
+    <div className={classes.root}>
+      <SectionTitle
+        title={`${isNewForm ? "New" : "Edit"} election candidate`}
+      />
+      {displayedErrorComponent}
+      <form 
+        className={classes.form}
+        onSubmit={(e) => {
+          e.preventDefault();
+          e.stopPropagation();
+          void form.handleSubmit();
+        }}
+      >
+        <form.Field name="electionName">
```

Comment:
We could probably just hide this field? (I know we showed it before, but seems unnecessary)

Reply:
True, but I'm going to leave it just to get this deployed